hack/testdata/pod-with-large-name.yaml

```diff
@@ -8,4 +8,4 @@ metadata:
 spec:
   containers:
   - name: kubernetes-serve-hostname
-    image: gcr.io/google_containers/serve_hostname:v1.4
+    image: gcr.io/google_containers/serve-hostname-amd64:1.5
```

Comment:
we have moved all the test image to `kubernetes-e2e-test-images` repo, should use `gcr.io/kubernetes-e2e-test-images/serve-hostname-amd64:1.x`